src/yalexs_ble/lock.py

```diff
@@ -210,40 +213,56 @@ async def _handle_missing_characteristic(self, char_uuid: str) -> None:
         await client.clear_cache()
         raise BleakError(f"Missing characteristic {char_uuid}")
 
-    def _parse_state(self, state: bytes) -> Iterable[LockStateValue] | None:
+    def _parse_state(
```

Comment:
This function is getting almost unreadable. It needs to be refactored to have some named constants for the positions

Reply:
I agree. The use of `state` to describe things is now confounded a bit as well as there's a difference between the state that results here & is passed to callbacks and the state that's the sequence of bytes read from the session & consumed here.

State from the session includes the (unparsed) activity information as well as the lock state. State passed to the callbacks filters out the (parsed) activity.

Reply:
I cleaned it up